clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -370,9 +370,9 @@ private void warnIfPartitionerDeprecated() {
             MetricConfig metricConfig = new MetricConfig().samples(config.getInt(ProducerConfig.METRICS_NUM_SAMPLES_CONFIG))
                     .timeWindow(config.getLong(ProducerConfig.METRICS_SAMPLE_WINDOW_MS_CONFIG), TimeUnit.MILLISECONDS)
                     .recordLevel(Sensor.RecordingLevel.forName(config.getString(ProducerConfig.METRICS_RECORDING_LEVEL_CONFIG)))
-                    .tags(metricTags);
-            List<MetricsReporter> reporters = CommonClientConfigs.metricsReporters(clientId, config);
+                    .tags(metricTags);            
             this.clientTelemetryReporter = CommonClientConfigs.telemetryReporter(clientId, config);
```

Comment:
The NPE still happen if the error happens before initializing `clientTelemetryReporter`. Maybe we should add null check for `clientTelemetryReporter` in closing.

Reply:
True but I do not see how an exception can be emitted before (except OOM).
Then checking for null an Optional looks very strange. In that case I would rather remove the Optional or make it non final with a default empty value.

Reply:
> True but I do not see how an exception can be emitted before (except OOM).

you are right :)